relay.go

```diff
@@ -266,7 +255,7 @@ func (r *Relay) handleNewStream(s network.Stream) {
 }
 
 func (r *Relay) handleHopStream(s network.Stream, msg *pb.CircuitRelay) {
-	if !r.hop {
+	if !r.isAllowedToHop(s) {
```

Comment:
we need to add an error code specific to rejections (something like `HOP_REJECTED`) instead of confusing users trying to debug the issue.

Reply:
We don't need to create breaking change with other code base and older version just to be polite and print the right error message.

Reply:
This is not a breaking change, we are adding an error code (and we can update the spec for it).
We do _need_ this error code or else debugging will be a nightmare.
Basically, it's `CANT_SPEAK_REALY` if `r.hop` is false and `RELAY_REFUSED` if the filter rejects the request.

Reply:
Ok I'm doing this, I didn't though about the fact a new error will code will still be understand as an error even if not known.

Reply:
@vyzo When talk about `RELAY_REFUSED` are you talking about `STOP_RELAY_REFUSED` or a new `HOP_RELAY_REFUSED` ?

Reply:
`HOP_RELAY_REFUSED`.